### PR TITLE
Improve requirements for v7.0.0

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Certain requirements disappeared with v7, I also have a function to recover the wording to unify it between the CLI and the web version. Withdrawal of the requirement "checkMissingFiles"
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | Non-regression tests on the checklist

### CODE REVIEW

File: classes/UpgradeSelfCheck.php [hunk 1]
```diff
@@ -181,24 +132,133 @@ class UpgradeSelfCheck
...
+            self::MAX_EXECUTION_TIME_VALUE_INCORRECT => $this->getMaxExecutionTime() > 0 && $this->getMaxExecutionTime() < 30,
+            self::APACHE_MOD_REWRITE_DISABLED => !$this->isApacheModRewriteEnabled(),
+            self::NOT_LOADED_PHP_EXTENSIONS_LIST_NOT_EMPTY => !empty($this->getNotLoadedPhpExtensions()),
+            self::NOT_EXIST_PHP_FUNCTIONS_LIST_NOT_EMPTY => !empty($this->getNotExistsPhpFunctions()),
+            self::MEMORY_LIMIT_INVALID => !$this->isMemoryLimitValid(),
+            self::PHP_FILE_UPLOADS_CONFIGURATION_DISABLED => !$this->isPhpFileUploadsConfigurationEnabled(),
+            self::KEY_GENERATION_INVALID => !$this->checkKeyGeneration(),
+            self::NOT_WRITING_DIRECTORY_LIST_NOT_EMPTY => !empty($this->getNotWritingDirectories()),
+            self::SHOP_VERSION_NOT_MATCHING_VERSION_IN_DATABASE => !$this->isShopVersionMatchingVersionInDatabase(),
+        ];
```
Comment (Quetzacoalt91) [inline classes/UpgradeSelfCheck.php:208]: If we're supposed to return only display errors, this method could return only the detected issues. Something OK would not appear.

And if you only return the issues, you could generate at this time the translated message? By doing so you would call trans only when necessary.

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by M0rgan01** [comment]
@Hlavtox Yes it is voluntary. This verification is already done by PrestaShop itself since PrestaShop 1.7, so we are dropping it. There will be problems before you even get to the module screen

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by ga-devfront — APPROVED** [review]

File: classes/UpgradeSelfCheck.php [hunk 2]
```diff
@@ -181,24 +132,209 @@ class UpgradeSelfCheck
...
+
+    /**
+     * @throws Exception
+     *
+     * @return array<int, bool>
+     */
+    public function getErrors(): array
+    {
+        $errors = [
+            self::PHP_REQUIREMENTS_INVALID => $this->getPhpRequirementsState() == self::PHP_REQUIREMENTS_INVALID,
```
Comment (Quetzacoalt91) [inline classes/UpgradeSelfCheck.php:191]:
```suggestion
            self::PHP_REQUIREMENTS_INVALID => $this->getPhpRequirementsState() === self::PHP_REQUIREMENTS_INVALID,
```

File: classes/UpgradeSelfCheck.php [hunk 3]
```diff
@@ -181,24 +132,209 @@ class UpgradeSelfCheck
...
+    }
+
+    /**
+     * @return array<int, bool>
+     */
+    public function getWarnings(): array
+    {
+        $warnings = [
+            self::MODULE_VERSION_IS_OUT_OF_DATE => !$this->isModuleVersionLatest(),
+            self::PHP_REQUIREMENTS_UNKNOWN => !$this->getPhpRequirementsState() == self::PHP_REQUIREMENTS_UNKNOWN,
```
Comment (Quetzacoalt91) [inline classes/UpgradeSelfCheck.php:220]:
```suggestion
            self::PHP_REQUIREMENTS_UNKNOWN => $this->getPhpRequirementsState() === self::PHP_REQUIREMENTS_UNKNOWN,
```

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by Hlavtox — CHANGES_REQUESTED** [review]
@M0rgan01 Please check this - https://github.com/PrestaShop/autoupgrade/pull/561

I specifically implemented the session check inside this module, because the core check was faulty.

**Comment by M0rgan01** [comment]
@Hlavtox While waiting to investigate the subject of PHP sessions, I am removing it from the PR so as not to block it

**Comment by Quetzacoalt91** [comment]
> @M0rgan01 Please see this - #561
> 
> I specifically implemented the session check inside this module, because the core check was faulty. I made the change to fix this issue - [PrestaShop/PrestaShop#30658](https://github.com/PrestaShop/PrestaShop/issues/30658).

@Hlavtox considering all the versions currently supported by the autoupgrade module require sessions to work properly, is this check still relevant in the module before an update?
I guess that if sessions don't work properly, the installation would have detected it, or in the worst case an error would have been thrown while using the backoffice or something?

**Comment by Hlavtox** [comment]
@Quetzacoalt91 Well, the problem was that even though sessions worked, the check present in the core did not.

So, if you want to completely skip this check, we can remove it, because if he go to the screen, the sessions work.
But, if you want to use the CORE check, it won't work for pre 8.0 cores, I fixed the issue for 8.0.x - https://github.com/PrestaShop/PrestaShop/pull/31241

**Review by AureRita — APPROVED** [review]
Hi @M0rgan01 

Thank you for your PR, I tested it and it seems to works as you can see : 

image

Tested from : 
1.7.7.8 to 8.1.7 
8.0.5 to 8.1.7 
8.1.7 to 9.0.0
8.1.5 to 9.0.0
8.0.5 to 9.0.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

File: classes/UpgradeSelfCheck.php [hunk 4]
```diff
@@ -181,24 +132,213 @@ class UpgradeSelfCheck
...
+                    'The following PHP extension%s need%s to be installed:',
+                    [$phpExtensionsCount > 1 ? 's' : '', $phpExtensionsCount === 1 ? 's' : '']
+                );
+
+            case self::NOT_EXIST_PHP_FUNCTIONS_LIST_NOT_EMPTY:
+                $phpFunctionsCount = count($this->getNotExistsPhpFunctions());
+
+                return $this->translator->trans(
+                    'The following PHP function%s need%s to be allowed:',
+                    [$phpFunctionsCount > 1 ? 's' : '', $phpFunctionsCount === 1 ? 's' : '']
```
Comment (Hlavtox) [inline classes/UpgradeSelfCheck.php:294]: This doesn't work for other languages. :-)

**Comment by Quetzacoalt91** [comment]
I've heard of this feature, but I wasn't sure it can be used. Seeing the opened PR seems to confirm there is no such thing on the Core? :/
We'll check again, thanks for the feedback in the meantime.

**Comment by Hlavtox** [comment]
@Quetzacoalt91 Symfony supports it out of the box, and it worked when I tried it, but it would require some minor tweaks. :-)

It's the proper way to do plurals in all languages.
Example:
- English:
  - 0 items
  - 1 item
  - 2 items
  - 3 items
  - 4 items etc.
- Czech:
  - 0 položek
  - 1 položka
  - 2 položky
  - 3 položky
  - 4 položky
  - 5 položek
  - 6 položek etc.

**Pull request merged**